codegen/src/types/substitutes.rs

```diff
@@ -108,48 +108,77 @@ impl TypeSubstitutes {
         }
     }
 
+    /// Only insert the given substitution if a substitution at that path doesn't
+    /// already exist.
+    pub fn insert_if_not_exists(
+        &mut self,
+        source: syn::Path,
+        target: AbsolutePath,
+    ) -> Result<(), CodegenError> {
+        let (key, val) = TypeSubstitutes::parse_path_substitution(source, target.0)?;
```

Comment:
Ok, this fails only if the parsing fails.

Can't we throw a warning or something if `insert` is ignored i.e, the given key already exist?

Reply:
I didn't bother only because this function is sortof dangling at the moment, but the point of it will be to override the RuntimeCall enum only if the user hasn't already overridden it via some type substitution of their own or whatever (which is necessary because we don't know where it lives at the point of setting up the default substitutions.. although maybe it could and I'll need to look into that).

Maybe it's best just to remove this function for now!